traffic_ops/traffic_ops_golang/server/put_status.go

```diff
@@ -121,7 +121,7 @@ func UpdateStatusHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if *status.Name == tc.CacheStatusAdminDown.String() || *status.Name == tc.CacheStatusOffline.String() {
+	if *status.Name != tc.CacheStatusOnline.String() && *status.Name != tc.CacheStatusReported.String() {
```

Comment:
So if I make a server have the status ONLINE1 or ONLINE_BUT_NOT_IN_SERVICE I then have to enter a reason why my ostensibly online server is offline?

Reply:
yes, unless we made "offline reason" always optional (never required as it is now)

Reply:
ONLINE1 or ONLINE_BUT_NOT_IN_SERVICE is not considered "online" from TC's perspective :)

Reply:
well i guess it could be for some servers :) i.e. TP or TO

Reply:
Server statuses should not be immutable. It's one of a million places where we hijacked a concept to provide arbitrary informative tagging, which is better provided by [actual tags](https://github.com/apache/trafficcontrol/pull/4819).

Reply:
Yeah, i get it but that ship has sailed (the immutable part), we already have many "offline" status types in production (i.e. TEST, PRE_PROD, DECOM, etc) that are all "offline" yet you can't add an offline reason to them.

Reply:
That can be retconned:

1. Implement tags
2. Migrate existing, unrecognized statuses to tags
3. Apply those tags on the servers that use those statuses
4. Convert all unknown statuses to "OFFLINE"
5. Eliminate the concept of a "Status" as an ATC object, use constant strings only.